hal/ia32/_interrupts.S

```diff
@@ -244,18 +221,27 @@ hal_cpuReschedule:
 	pushl %eax             /* push far cs:address of return */
 	xorl %eax, %eax
 	call interrupts_pushContext
-	_INTERRUPTS_MULTILOCKSET %eax
 	mov %esp, %eax
 	pushl $0               /* n */
 	pushl %eax             /* cpu context */
 	pushl $0               /* arg */
-	call threads_schedule
 
-	cli                    /* hal_cpuDisableInterrupts() */
+	movl 8(%ebp), %ecx     /* ecx := spinlock */
+	testl %ecx, %ecx
+	jz .Lnospinlock
```

Comment:
Hmm, ok, it should work as is, anyway it would be better to have just one check to avoid spaghetti code. But it shouldn't make it not work properly, so that's a lower priority matter